### PR TITLE
Update docs to v1.4

### PR DESCRIPTION
**Pull request by Zeniuus**
PR for updating docs to follow API v1.4.

Still in progress.

### CODE REVIEW

**Comment by Zeniuus** [comment]
I schemed through all docs and updated what I found deprecated.

Please check if I missed anything or give me comments about wording! (since I'm not confident with my English :( )

File: docs/common-api/convention.rst [hunk 1]
```diff
@@ -93,6 +93,10 @@ Therefore, all API query URLs are prefixed with the major revision, such as ``/v
 Minor releases may introduce new parameters and response fields but no URL changes.
 Accessing unsupported major revision returns HTTP 404 Not Found.
 
+.. versionchanged:: v3.20170615
+   Version prefix in API queries are deprecated. (Yet still supported currently)
+   For example, now users should call ``/kernel/create`` rather than ``/v2/kernel/create``.
+
```
Comment (achimnol) [inline docs/common-api/convention.rst:99]: 👍

File: docs/common-api/objects.rst [hunk 2]
```diff
@@ -218,7 +218,7 @@ Execution Result Object
    * - ``status``
      - ``enum[str]``
 
-     - One of ``"continued"``, ``"waiting-input"``, ``"finished"``, or ``"build-finished"``.
+     - One of ``"continued"``, ``"waiting-input"``, ``"finished"``, ``"build-finished"``, or ``"exec-timeout"``.
        See more details at :ref:`code-execution-model`.
```
Comment (achimnol) [inline docs/common-api/objects.rst:223]: Shouldn't we add `clean-finished` which you have added for lablup/backend.ai-agent#74 ?

Comment (Zeniuus) [reply docs/common-api/objects.rst:223]: I thought that should go later, not this PR, because this PR is for 1.4 branch...

Comment (Zeniuus) [reply docs/common-api/objects.rst:223]: ...and I found this PR merges into master! I'll add `clean-finished` :D

**Comment by Zeniuus** [comment]
@achimnol Updated!

**Comment by Zeniuus** [comment]
@achimnol

**Pull request merged**